packages/vue-renderer/src/spa-meta.js

```diff
@@ -30,8 +30,9 @@ export default class SPAMetaRenderer {
     return vm.$meta().inject()
   }
 
-  async render({ url = '/', req = {} }) {
-    const cacheKey = `${req.modernMode ? 'modern:' : 'legacy:'}${url}`
+  async render({ url = '/', req = {}, _generate }) {
+    const modern = req.modernMode || _generate
```

Comment:
How `nuxt generate` implies modern mode is enabled?

Reply:
`req.modernMode` is set through middleware which is not called in `generate`.

With this fix, the behaviour is same as `vue-cli` doing

`modulepreload`: all modern js, only in modern browsers
`<script type='module'>`: all modern js, only in modern browsers
`<script nomodule>`: all legacy js, only in legacy browsers

Reply:
I agree on this, we want to make sure he built with the `modern` mode too no?

Reply:
Oh, sorry, I missed the `modern` is not enabled case

Reply:
@pi0 @Atinux I added `options.modern` checking, only generate `modern` resources in modern mode.